### DISCUSSION TITLE
How do you incorporate framework-specific element wrappers into UI package?

### DISCUSSION THREAD

**Discussion by GabenGar** [vercel/turborepo#9251] [Help] [answered]
### Summary

The most obvious example is [permutations](https://reactrouter.com/en/main/components/link) of [`<a>`](https://nextjs.org/docs/app/api-reference/components/link) [element](https://remix.run/docs/en/main/components/link). At least the styling of links should be governed from the single place, aka UI package. But their props shapes are all over the place and doing conditional rendering in UI package may confuse the bundler and include other frameworks into dependency tree. So UI package has to settle only for wrappers for vanilla elements. 
I presume, since framework-specific wrappers have to be declared somewhere, as long as there is only one app with a framework just create those wrappers in the app itself. But then vanilla wrappers from UI package cannot be used because aforementioned components are provided as separate elements. Some sort of component factory which applies classnames to a provided component maybe?

**Comment by anthonyshew** [comment, marked as answer]
Hey, we've written a doc about binding to frameworks here! https://turbo.build/repo/docs/guides/frameworks/framework-bindings

**Comment by GabenGar** [comment]
How does this work in IDE? Will I just see all possible globals from all frameworks while working in UI package?

**Reply by anthonyshew** [reply]
It depends how you architect. If you add multiple frameworks as peer dependencies to one package, yes, you would.
